### PR TITLE
Add abstract class for golden file testing

### PR DESCRIPTION
**Pull request by holounic**
Since JSONs are used for testing fetchers, it's nice to have convenient tool to easily access these input and expected outputs and store them in files instead of having long hardcoded Strings. 
The ```AbstractJsonTestInfo``` class contains common methods of the already implemented ```GitHubJsonTestInfo``` and ```BuildBotJsonTestInfo``` which will be implemented in the future PR. The last mentioned class is required to test a BuildBot fetcher.

### CODE REVIEW

**Review by MagoDelBlocco — APPROVED** [review]
LGTM, left a few nits

File: src/test/java/com/google/graphgeckos/dashboard/GitHubJsonTestInfo.java [hunk 1]
```diff
@@ -49,22 +37,12 @@ private void assignExpectedValues(String[] expected) {
     repositoryLink = expected[3];
   }
 
+
   /**
-   * @param testName Common part of the input and output file names. E.g the test name "real_json" is expected
-   *                to have input_real_json.txt file with input json and output_real_json.txt with expected output.
+   * {@inheritDoc}
    */
   public GitHubJsonTestInfo(String testName) {
```
[the remark this answers is not in the document]
Comment (holounic) [reply src/test/java/com/google/graphgeckos/dashboard/GitHubJsonTestInfo.java]: Done!

File: src/test/java/com/google/graphgeckos/dashboard/AbstractJsonTestInfo.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+   * @param testName Common part of the input and output file names. E.g the test name "real_json" is expected
+   *                to have input_real_json.txt file with input json and output_real_json.txt with expected output.
+   */
+  public AbstractJsonTestInfo(String testName) {
+    try {
+      content = readFile(getPath() + INPUT_PREFIX + testName + FILE_FORMAT);
+    } catch (IOException e) {
+      System.out.println("File not found:" + e.getMessage());
+    }
+    try {
```
[the remark this answers is not in the document]
Comment (holounic) [reply src/test/java/com/google/graphgeckos/dashboard/AbstractJsonTestInfo.java:36]: Done!

**Review by krasimir-google — APPROVED** [review]
LGTM!

This is nice!

**Pull request merged**